core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -1441,12 +1442,15 @@ private Optional<UfsStatus> syncPersistDirectory(InodeDirectoryView dir, boolean
       MkdirsOptions mkdirsOptions =
           MkdirsOptions.defaults(Configuration.global()).setCreateParent(false)
           .setOwner(dir.getOwner()).setGroup(dir.getGroup()).setMode(new Mode(dir.getMode()));
-      if (!ufs.mkdirs(ufsUri, mkdirsOptions)) {
+      if (isMetadataLoad || !ufs.mkdirs(ufsUri, mkdirsOptions)) {
```

Comment:
it's quite weird why a `mount` or `loadMetadata` will call `syncPersistDirectory()`. I'm not sure how that happened. What is the call stack when this happens?

Reply:
Thanks for helping review the pull request! The stack trace shows below.

```
java.lang.Throwable
	at alluxio.master.file.meta.InodeTree.syncPersistDirectory(InodeTree.java:1423)
	at alluxio.master.file.meta.InodeTree.syncPersistExistingDirectory(InodeTree.java:1343)
	at alluxio.master.file.meta.InodeTree.createPath(InodeTree.java:966)
	at alluxio.master.file.DefaultFileSystemMaster.createDirectoryInternal(DefaultFileSystemMaster.java:2762)
	at alluxio.master.file.InodeSyncStream.loadDirectoryMetadataInternal(InodeSyncStream.java:1384)
	at alluxio.master.file.InodeSyncStream.loadDirectoryMetadata(InodeSyncStream.java:1304)
	at alluxio.master.file.InodeSyncStream.loadMetadata(InodeSyncStream.java:1051)
	at alluxio.master.file.InodeSyncStream.loadMetadataForPath(InodeSyncStream.java:1019)
	at alluxio.master.file.InodeSyncStream.syncInodeMetadata(InodeSyncStream.java:751)
	at alluxio.master.file.InodeSyncStream.syncInternal(InodeSyncStream.java:495)
	at alluxio.master.file.InodeSyncStream.sync(InodeSyncStream.java:413)
	at alluxio.master.file.DefaultFileSystemMaster.syncMetadata(DefaultFileSystemMaster.java:4100)
	at alluxio.master.file.TxFileSystemMaster.syncMetadata(TxFileSystemMaster.java:347)
	at alluxio.master.file.DefaultFileSystemMaster.listStatus(DefaultFileSystemMaster.java:1114)
	at alluxio.master.file.TxFileSystemMaster.listStatus(TxFileSystemMaster.java:915)
	at alluxio.master.file.FileSystemMasterClientServiceHandler.lambda$listStatus$10(FileSystemMasterClientServiceHandler.java:311)
	at alluxio.RpcUtils.callAndReturn(RpcUtils.java:129)
	at alluxio.master.file.FileSystemMasterClientServiceHandler.listStatus(FileSystemMasterClientServiceHandler.java:309)
	at alluxio.grpc.FileSystemMasterClientServiceGrpc$MethodHandlers.invoke(FileSystemMasterClientServiceGrpc.java:3867)
```